Prox/Prox/PlaceDetails/FilterViewController.swift

```diff
@@ -0,0 +1,193 @@
+        view.addSubview(container)
+
+        placeCountLabel.font = UIFont.systemFont(ofSize: 20)
+        container.addSubview(placeCountLabel)
+
+        let closeButton = UIButton()
+        closeButton.setImage(#imageLiteral(resourceName: "button_dismiss"), for: .normal)
+        closeButton.addTarget(self, action: #selector(didPressClose), for: .touchUpInside)
+        view.addSubview(closeButton)
+
+        let spacing: CGFloat = 16
+        stackView.axis = .vertical
+        stackView.spacing = spacing
+        stackView.alignment = .center
+        stackView.distribution = .fillEqually
+        container.addSubview(stackView)
+
+        showFilterConstraints += [
+            container.topAnchor.constraint(equalTo: view.topAnchor),
+            placeCountLabel.topAnchor.constraint(equalTo: topLayoutGuide.bottomAnchor, constant: spacing),
+        ]
+
+        hideFilterConstraints += [
+            container.bottomAnchor.constraint(equalTo: view.topAnchor),
+        ]
+
+        let constraints = [
+            background.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            background.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            background.topAnchor.constraint(equalTo: view.topAnchor),
+            background.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+
+            container.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            container.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+
+            placeCountLabel.centerXAnchor.constraint(equalTo: container.centerXAnchor),
+
+            closeButton.centerYAnchor.constraint(equalTo: placeCountLabel.centerYAnchor),
+            closeButton.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -spacing),
+
+            stackView.topAnchor.constraint(equalTo: placeCountLabel.bottomAnchor, constant: spacing),
+            stackView.bottomAnchor.constraint(equalTo: container.bottomAnchor, constant: -spacing),
+            stackView.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+        ]
+
+        NSLayoutConstraint.activate(constraints + hideFilterConstraints, translatesAutoresizingMaskIntoConstraints: false)
+    }
+
+    var placeCount: Int = 0 {
+        didSet {
+            placeCountLabel.text = String(format: Strings.filterView.placeCount, placeCount)
+        }
+    }
+
+    @objc private func didToggleFilter(sender: FilterButton) {
+        let filter = filters[sender.tag]
+        filter.enabled = !filter.enabled
+        sender.isSelected = filter.enabled
+        delegate?.filterViewController(self, didUpdateFilters: filters)
+    }
+
+    @objc private func didPressClose() {
+        dismiss(animated: true, completion: nil)
+    }
+
+    override func dismiss(animated flag: Bool, completion: (() -> Void)? = nil) {
+        super.dismiss(animated: flag, completion: completion)
+        delegate?.filterViewController(self, didDismissWithFilters: filters)
```

Comment:
I'm concerned that `dismiss` can be used to dismiss a ViewController other than self – like a UIAlertViewController.